Added tests, locators, README.

SpessComrade opened the pull request:
Добавил тесты, локаторы, файл с фикстурами и  README

SpessComrade committed:
Added tests, locators, README.

SpessComrade commented:
conftest создал в папке с тестами, тк. слышал, что в Pycharm периодически багается запуск фикстур, если они не расположены в папке с исполняемыми файлами.

SpessComrade committed:
Updated README, added requirements.txt.

SpessComrade commented:
Обновил Ридми, забыл добавить спецификацию проекта. Также добавил requirements.txt

AlexKitIt commented on Locators.py:
Нужно исправить: именование пакетов, модулей, методов в питоне ведется в snake_case, нейминг необходимо исправить в соответствии с этим правилом

AlexKitIt commented on Locators.py:
```diff
@@ -0,0 +1,32 @@
+    #Кнопки
+    REGISTRATION_BUTTON = [By.XPATH, "//button[text()='Зарегистрироваться']"]
+    PAGE_LOGIN_BUTTON = [By.XPATH, '//*[@id="root"]/div/main/section[2]/div/button']
+    GOTO_ACCOUNT_BUTTON = [By.XPATH, '//*[@id="root"]/div/header/nav/a']
+    REGISTRATION_N_RECOVERY_GOTO_LOGIN_BUTTON = [By.XPATH, '//*[@id="root"]/div/main/div/div/p/a'] #Раньше тут были 2 разных локатора на кнопки перехода к логину из регистрации и из забытого пароля, но какой-либо разницы в вообще всех известных мне локаторах обнаружено не было.
```
Нужно исправить: здесь и далее: использование порядкового номера элемента в dom является весьма нестабильным способом поиска, необходимо привязываться к атрибутам конкретного элемента
Кроме того ходить от root и использовать абсолютные пути нельзя - очень ненадежно, добавится на пути новый элемент - тест ляжет. Все подобные локаторы необходимо переписать

AlexKitIt commented on tests/conftest.py:
```diff
@@ -0,0 +1,21 @@
+@pytest.fixture
+def driver():
+    driver = webdriver.Chrome()
+    driver.get("https://stellarburgers.nomoreparties.site/")
```
Можно лучше: урлы стоит хранить отдельно, они постоянно переиспользуются, для них стоит создать модуль urls или отдельный класс в data

AlexKitIt commented on tests/conftest.py:
```diff
@@ -0,0 +1,21 @@
+def login(driver):
+    driver.get("https://stellarburgers.nomoreparties.site/login")
+
+    driver.find_element(By.XPATH, '//label[ text()="Email" ]/parent::div/input').send_keys('romanrubtsov_14@gmail.com')
```
Нужно исправить: здесь и далее: все локаторы хранятся отдельно от тестов в модуле locators, а тут только переменные, в которые они были записаны

AlexKitIt commented on tests/tests_login.py:
```diff
@@ -0,0 +1,61 @@
+        WebDriverWait(driver, 10).until(expected_conditions.presence_of_element_located(
+            TestLocators.PAGE_LOGIN_BUTTON)).click()
+
+        driver.find_element(*TestLocators.EMAIL_INPUT).send_keys('romanrubtsov_14@gmail.com')
+        driver.find_element(*TestLocators.PASSWORD_INPUT).send_keys('q1w2e3r4')
```
Можно лучше: тестовые данные стоит хранить отдельно в data модуле, так поддерживать и переиспользовать будет проще, в тестах и тестовых классах их держать не стоит

AlexKitIt commented on tests/tests_registration.py:
```diff
@@ -0,0 +1,48 @@
+        driver.find_element(*TestLocators.PASSWORD_INPUT).send_keys('q1w2e3r4')
+
+        driver.find_element(*TestLocators.REGISTRATION_BUTTON).click()
+        time.sleep(1)
```
Нужно исправить: здесь и далее: слипов и неявных ожиданий быть не должно. Необходимо использовать expected_conditions и встраивать ожидания на нем.

SpessComrade committed:
Обновлены файлы тестов (методы переименованы под snake_case, урлы вынесены в отдельный модуль, удалены слипы, оптимизированы локаторы)

Отрефакторен locators для соответствия snake_case

Добавлен модуль urls для хранения ссылок

Отредактирован README (убраны пути локаторов, добавлено описание ссылок)
